src/plugin/PluginManager.php

```diff
@@ -429,9 +433,26 @@ public function isPluginEnabled(Plugin $plugin) : bool{
 	}
 
 	/**
-	 * @param Plugin $plugin
+	 * @param PluginLoadOrder $type
 	 */
-	public function enablePlugin(Plugin $plugin) : void{
+	public function enablePlugins(PluginLoadOrder $type) : void{
+		if($this->scanLoaded){
+			$this->scanEnabled = true;
+		}
+
+		foreach($this->plugins as $plugin){
+			if(!$plugin->isEnabled() and $plugin->getDescription()->getOrder()->equals($type)){
+				$this->enablePlugin($plugin);
+			}
+		}
+
+		if($type->equals(PluginLoadOrder::POSTWORLD())){
+			$this->server->getCommandMap()->registerServerAliases();
+			DefaultPermissions::registerCorePermissions();
+		}
```

Comment:
I think this block should be located in the Server startup routine. I have no idea why it was here to begin with, and it certainly doesn't make sense to reinitialize all this crap if something decides to add an extra plugin.

Perhaps this change should be done separately, but I'm logging it here for posterity since you've touched the code.